thunder/tests/test_nvfuser_remat.py

```diff
@@ -333,8 +333,9 @@ def test_find_cut_dropout(executor, device, _):
     # Note cut[1]/producer.output[0] is the boolean mask for dropout. It should
     # be chosen over the float32 mask. See this issue: "The Recomputation
     # Algorithm on Dropout choses a float32 mask to save"
-    assert cut[1] == producer.output[0].name
-    assert cut[2] == producer.output[1].name
+    producer_output_names = tuple(o.name for o in producer.output)
+    assert cut[1] in producer_output_names
+    assert cut[2] in producer_output_names
```

Comment:
Why does the order change?

Reply:
I **guess** non-determinism (e.g. iterating an unordered map) that we could have hit earlier but got lucky? But if these functions were designed to be deterministic then something broke that property and needs to be fixed instead.

Reply:
> I think this test flakiness only indicates a hidden problem with the non-deterministic naming of variables.

Never mind -- I realized you sensed the same problem.